src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreJob.cs

```diff
@@ -499,6 +507,17 @@ private async Task RestoreAsync(bool forceRestore, RestoreOperationSource restor
                                 {
                                     if (isRestoreSucceeded)
                                     {
+                                        bool shouldDisplayVulnerabilitiesInfoBar = LogUtility.AreVulnerabilitiesInRestoreSummaries(restoreSummaries);
+
+                                        if (shouldDisplayVulnerabilitiesInfoBar)
+                                        {
+                                            await _infoBarService.Value.Show(t);
+                                        }
+                                        else if (_infoBarService.IsValueCreated) // if the InfoBar was created and all vulnerabilities are fixed, hide it.
```

Comment:
This seems fine since it probably needs to switch to the UI thread every time to determine if the infobar is visible or not.
I will say that the code is a bit subtle. It's basically saying, "if we ever created the infobar, then let's see if we need to hide it on every restore for the remainder of this VS session".